### PR TITLE
Other metrics than accuracy

### PR DESCRIPTION
**Pull request by dafnevk**
This solves issue #128: it is now possible to use other metrics that Keras knows (https://keras.io/metrics/) for evaluating the model on the valdiation set. The visualization is also adapted, although this could be done nicer.

### CODE REVIEW

File: mcfly/find_architecture.py [hunk 1]
```diff
@@ -65,12 +68,17 @@ def train_models_on_samples(X_train, y_train, X_val, y_val, models,
...
     histories = []
-    val_accuracies = []
+    val_metrics = []
     val_losses = []
     for i, (model, params, model_types) in enumerate(models):
         if verbose:
             print('Training model %d' % i, model_types)
+        model_metrics = [get_metric_name(name) for name in model.metrics]
+        if metric_name not in model_metrics:
+            raise ValueError('Invalid metric, the model should be compiled with the same metric!')
```
Comment (cwmeijer) [inline mcfly/find_architecture.py]: Error message is vague as it is. The message should include the currently selected metric at least. Also, I would prefer a period instead of an exclamation mark.
